src/ds/ring_buffer_types.h

```diff
@@ -96,6 +96,11 @@ namespace ringbuffer
 
     virtual WriteMarker write_bytes(
       const WriteMarker& marker, const uint8_t* bytes, size_t size) = 0;
+
+    virtual std::optional<size_t> get_max_message_size()
+    {
+      return std::nullopt;
+    }
```

Comment:
I don't think we need an optional here; there's a sensible value for every implementation of `AbstractWriter`, so this could just be `virtual size_t get_max_message_size() = 0;`?

Then `ringbuffer::Writer` would override it to return `Const:max_size()` (enforced in prepare), and `NonBlockingWriter` would override it to return `underlying_writer->get_max_message_size()`.

Reply:
Good point, I'd missed that we checked for the maximum size in the vanilla ring buffer implementation.